### PR TITLE
Windows: Build lua with 5.2 compatibility

### PR DESCRIPTION
**Pull request by sunjayBhatia**
Commit Message: Windows: Build lua with 5.2 compatibility
Additional Description: As per http://lua-users.org/wiki/MetatableEvents the __pairs metamethod is not available until lua 5.2, luajit/moonjit build lua 5.1. This define was set on Linux/Mac but not Windows.
Risk Level: Low
Testing: Enables 3 failing lua tests on Windows, checked locally on Windows
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
cc @envoyproxy/windows-dev

File: bazel/foreign_cc/luajit.patch [hunk 1]
```diff
@@ -59,7 +59,7 @@ EOF
  
  @setlocal
 -@set LJCOMPILE=cl /nologo /c /O2 /W3 /D_CRT_SECURE_NO_DEPRECATE /D_CRT_STDIO_INLINE=__declspec(dllexport)__inline
-+@set LJCOMPILE=cl /nologo /c /W3 /D_CRT_SECURE_NO_DEPRECATE /D_CRT_STDIO_INLINE=__declspec(dllexport)__inline
++@set LJCOMPILE=cl /nologo /c /W3 /D_CRT_SECURE_NO_DEPRECATE /D_CRT_STDIO_INLINE=__declspec(dllexport)__inline /DLUAJIT_ENABLE_LUA52COMPAT
```
Comment (dio) [inline bazel/foreign_cc/luajit.patch:62]: Sorry, just curious, why we remove `/O2` here?

Comment (sunjayBhatia) [reply bazel/foreign_cc/luajit.patch:62]: `/O2` placed there was being applied to to debug and release build modes, it was removed here to enable us to target it to just release mode and disable optimizations entirely in debug

cc @wrowe

Comment (wrowe) [reply bazel/foreign_cc/luajit.patch:62]: You'll find /O2 is still applied appropriately later in the script, @dio

Comment (dio) [reply bazel/foreign_cc/luajit.patch:62]: Got it. Thanks, @wrowe.

**Comment by dio** [comment]
@sunjayBhatia this is looks good. However, we need to merge main to pick up a fix.

**Comment by wrowe** [comment]
@dio done, thanks. (Sunjay is on PTO)

**Review by dio — APPROVED** [review]
Thank you!

**Review by wrowe — APPROVED** [review]
LGTM

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**